src/main/scala/com/stratio/khermes/cluster/supervisor/NodeSupervisorActor.scala

```diff
@@ -36,44 +38,45 @@ import scala.util.Try
  * Supervisor that will manage a thread that will generate data along the cluster.
  * @param config with all needed configuration.
  */
-class KhermesSupervisorActor(implicit config: Config) extends Actor with ActorLogging {
+class NodeSupervisorActor(implicit config: Config) extends Actor with ActorLogging {
 
   import DistributedPubSubMediator.Subscribe
 
   val mediator = DistributedPubSub(context.system).mediator
   mediator ! Subscribe("content", self)
 
-  var khermesExecutor: Option[KhermesExecutor] = None
+  var khermesExecutor: Option[NodeExecutorThread] = None
```

Comment:
same as above comment about avoiding vars. review this [post](https://scalerablog.wordpress.com/2017/03/20/scalera-tip-keep-your-actors-state-with-no-var-at-all/)